CRM/Financial/BAO/Payment.php

```diff
@@ -59,12 +59,7 @@ public static function create($params) {
 
     $isPaymentCompletesContribution = self::isPaymentCompletesContribution($params['contribution_id'], $params['total_amount']);
 
-    // For legacy reasons Pending payments are completed through completetransaction.
-    // @todo completetransaction should transition components but financial transactions
-    // should be handled through Payment.create.
-    $isSkipRecordingPaymentHereForLegacyHandlingReasons = ($contributionStatus == 'Pending' && $isPaymentCompletesContribution);
-
-    if (!$isSkipRecordingPaymentHereForLegacyHandlingReasons && $params['total_amount'] > 0) {
```

Comment:
Just a note that changing the contribution status requires making financial transactions to record that. I assume that this change is not making it so that no longer happens.

Reply:
no the change is making it so that the payment.create action records financial transactions regardless of whether the payment completes the contribution or not - it was leaving that to the later Contribution.create only when the payment completed the transaction